src/main/java/com/example/airbnbapi/api/Controller.java

```diff
@@ -18,25 +17,39 @@
 
 
     @Autowired
-    public Controller( ServiceFactory serviceFactory) {
+    public Controller(ServiceFactory serviceFactory) {
 
         this.serviceFactory = serviceFactory;
 
     }
 
     @GetMapping(path = "/{type}")
-    public  Media[] getAllMedia (@PathVariable("type") MediaType type) {
-      return   serviceFactory.getServiceByType(type).getItems();
+    public ResponseEntity<Media[]> getAllMedia(@PathVariable("type") MediaType type) {
+
+        Media[] items = serviceFactory.getServiceByType(type).getItems();
+        if (items != null) {
```

Comment:
So as discussed, items will never be null as we control the json files; probably best to remove the if clause entirely